pqarrow/arrow.go

```diff
@@ -94,15 +104,54 @@ func contiguousParquetRowGroupToArrowRecord(
 	pool memory.Allocator,
 	rg parquet.RowGroup,
 	projections []logicalplan.ColumnMatcher,
+	filterExpr logicalplan.Expr,
+	distinctColumns []logicalplan.ColumnMatcher,
 ) (arrow.Record, error) {
 	s := rg.Schema()
-
 	children := s.ChildNames()
+
+	if len(distinctColumns) == 1 && filterExpr == nil {
+		// We can use the faster path for a single distinct column by just
+		// returning its dictionary.
+		fields := make([]arrow.Field, 0, 1)
+		cols := make([]array.Interface, 0, 1)
```

Comment:
```suggestion
		cols := make([]arrow.Array, 0, 1)
```
The Interface is just an alias to `arrow.Array`